src/main/kotlin/uk/gov/justice/digital/hmpps/createandvaryalicenceapi/service/dates/ReleaseDateService.kt

```diff
@@ -103,34 +103,24 @@ class ReleaseDateService(
 
   fun getLicenceStartDate(
     nomisRecord: PrisonerSearchPrisoner,
-    licenceKind: LicenceKind? = CRD,
-  ): LocalDate? = when (licenceKind) {
+    licenceKind: LicenceKind? = null,
+  ): LocalDate? = when (licenceKind ?: determineReleaseDateKind(nomisRecord.postRecallReleaseDate, nomisRecord.conditionalReleaseDate)) {
     HDC -> nomisRecord.homeDetentionCurfewActualDate
     PRRD -> nomisRecord.calculatePrrdLicenceStartDate()
     else -> calculateCrdLicenceStartDate(nomisRecord, iS91DeterminationService.isIS91Case(nomisRecord))
   }
 
-  fun getLicenceStartDates(
-    prisoners: List<PrisonerSearchPrisoner>,
-    prisonersToLicenceKinds: Map<String, LicenceKind>,
-  ): Map<String, LocalDate?> {
+  fun getLicenceStartDates(prisoners: List<PrisonerSearchPrisoner>): Map<String, LocalDate?> {
     val iS91BookingIds = iS91DeterminationService.getIS91AndExtraditionBookingIds(prisoners)
     return prisoners.associate {
-      it.prisonerNumber to when (prisonersToLicenceKinds[it.prisonerNumber]) {
+      it.prisonerNumber to when (determineReleaseDateKind(it.postRecallReleaseDate, it.conditionalReleaseDate)) {
         HDC -> it.homeDetentionCurfewActualDate
```

Comment:
A note for work that will need to be done if HDC cases are reintroduced to CVL:
Currently the HDC branch is unreachable here. It was previously too, because currently HDC cases are filtered out primarily by their `kind` not being determined for `NOT_STARTED` cases. If they are reintroduced fully, `determineReleaseDateKind` will need to be expanded, at which point this should be reachable.